overlays/haskell.nix

```diff
@@ -105,6 +105,11 @@ self: super: {
                 # The compiler referenced in the stack config
                 compiler = (stack-pkgs.extras hackage).compiler or (pkg-def hackage).compiler;
                 patchesModule = ghcHackagePatches.${compiler.nix-name} or {};
+                # Remove fake packages generated from stack keywords used in ghc-options
+                removeStackSpecial = module: if builtins.typeOf module == "set"
+                  then module // { packages = removeSpecialPackages (module.packages or {}); }
+                  else module;
+                removeSpecialPackages = ps: removeAttrs ps [ "$locals" "$targets" "$everything" ];
```

Comment:
You wish you had a `where` clause, don't you? 😏

Reply:
Yes, and also types and the `.` and `$` operators.